nixos/modules/services/x11/desktop-managers/xsession.nix

```diff
@@ -0,0 +1,33 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.xserver.desktopManager.xsession;
+
+in
+
+{
+  options = {
+
+    services.xserver.desktopManager.xsession.enable = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Enable ~/.xsession as a desktop manager.";
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    services.xserver.desktopManager.session = singleton
+      { name = "xsession";
+        start = ''
+          ${pkgs.gnome3.zenity}/bin/zenity --error --text 'The user must provide a ~/.xsession file containing session startup commands.' --no-wrap
```

Comment:
Using this program, would it bring in some undesirable deps for certain users?

Reply:
Yes, I wanted to use xdialog but that was seemingly not packaged. If you know of another similar package I'm all ears.

Reply:
We could use `xmessage` (though with a far less readable error message).

Reply:
There's xmessage but it's kinda ugly. Looking.

Reply:
Huh, looks like I commented that before refreshing :rofl:

Reply:
Seems xdialog also uses gtk so I'm not sure it matters much.

Reply:
I was thinking about `xmessage` when I said `xdialog`. I do agree this require a change to not bring gnome3 into scope.

Reply:
Here, being pretty is not a desired outcome. The outcome is to ensure that people relying on xsession have a solution, while at least showing *something* to other users explaining what's going on.

Reply:
Let's stick with zenity then (unless anyone have another good option).
I don't want to sacrifice readability.